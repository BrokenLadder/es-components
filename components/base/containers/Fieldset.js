import React from 'react';
import classnames from 'classnames';

function renderLegend(text, additionalLegendClasses) {
  const legendClasses = classnames('legend-style', additionalLegendClasses);
  return text ?
    <legend className={legendClasses}>{text}</legend> :
    null;
}

function Fieldset({
  additionalLegendClasses,
  legendText,
  children
}) {
  return (
    <fieldset>
      { renderLegend(legendText, additionalLegendClasses) }
      { children }
    </fieldset>
  );
}

Fieldset.propTypes = {
  /**
   * determine whether or not to add a legend and what text to display
   */
  legendText: React.PropTypes.string,
  /**
   * Additional classes to be applied to the legend element
   */
  additionalLegendClasses: React.PropTypes.string,
  children: React.PropTypes.element
};

export default Fieldset;
