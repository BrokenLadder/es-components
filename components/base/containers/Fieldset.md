Use a ``Fieldset`` component to group related fields.

### Fieldset example without a legend:
```
<Fieldset>
  <FormGroup>
    <label>
      <span className="label-style">First name:</span>
      <input type="text" />
    </label>
  </FormGroup>
  <FormGroup>
    <label>
      <span className="label-style">Middle initial:</span>
      <input type="text" />
    </label>
  </FormGroup>
  <FormGroup>
    <label>
      <span className="label-style">Last name:</span>
      <input type="text" />
    </label>
  </FormGroup>
</Fieldset>
```

### Fieldset with a legend

```
<Fieldset legendText="Please enter your name">
  <FormGroup>
    <label>
      <span className="label-style">First name:</span>
      <input type="text" />
    </label>
  </FormGroup>
  <FormGroup>
    <label>
      <span className="label-style">Middle initial:</span>
      <input type="text" />
    </label>
  </FormGroup>
  <FormGroup>
    <label>
      <span className="label-style">Last name:</span>
      <input type="text" />
    </label>
  </FormGroup>
</Fieldset>
```