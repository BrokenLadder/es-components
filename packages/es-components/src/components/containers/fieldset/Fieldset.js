import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import classnames from 'classnames';

const Legend = styled.legend`
  border: 0;
  border-bottom: 1px solid ${props => props.theme.colors.gray6};
  color: ${props => props.theme.colors.black};
  display: block;
  font-size: 27px;
  line-height: ${props => props.theme.sizes.baseLineHeight};
  margin: 0 0 25px 0;
  padding: 0;
  width: 100%;
`;

function renderLegend(content, legendClasses) {
  return content ? (
    <Legend className={classnames('es-fieldset__legend', legendClasses)}>
      {content}
    </Legend>
  ) : null;
}

const StyledFieldset = styled.fieldset`
  border: 0;
  margin: 0;
  padding: 0;
`;

function Fieldset({ legendClasses, legendContent, children }) {
  return (
    <StyledFieldset className="es-fieldset">
      {renderLegend(legendContent, legendClasses)}
      {children}
    </StyledFieldset>
  );
}

Fieldset.propTypes = {
  /** Determine whether or not to add a legend and what content to display  */
  legendContent: PropTypes.node,
  /** Additional classes to be applied to the legend element */
  legendClasses: PropTypes.string,
  children: PropTypes.node
};

export default Fieldset;
