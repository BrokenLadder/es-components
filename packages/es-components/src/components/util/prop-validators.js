import isParseableDate from './isParseableDate';

export function isPropValidDate(propsList, propName, componentName) {
  const date = propsList[propName];
  if (date) {
    if (!isParseableDate(date)) {
      return new Error(`The pre-selected date is not valid. Please enter a parseable date.
        See http://en.wikipedia.org/wiki/ISO_8601 for valid date formats.
      `);
    }
    return null;
  }
  return null;
}
