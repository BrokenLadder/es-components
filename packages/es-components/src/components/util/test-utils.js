import React from 'react';
import { ThemeProvider } from 'styled-components';
import viaTheme from 'es-components-via-theme';
import { render } from 'react-testing-library';

export function renderWithTheme(component) {
  return render(
    <ThemeProvider theme={viaTheme}>
      {React.cloneElement(component)}
    </ThemeProvider>
  )
}