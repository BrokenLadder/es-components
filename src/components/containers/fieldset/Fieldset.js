import React from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeProvider } from 'styled-components';

import defaultTheme from '../../theme/defaultTheme';

const Legend = styled.legend`
  border: 0;
  border-bottom: 1px solid ${props => props.theme.colors.grayDark};
  color: ${props => props.theme.colors.grayDark};
  display: block;
  font-size: ${props => props.theme.sizes.baseFontSize * 1.5}px;
  margin: 0 0 25px 0;
  padding: 0;
  width: 100%;
`;

function renderLegend(text, additionalLegendClasses) {
  return text ? (
    <Legend className={additionalLegendClasses}>{text}</Legend>
  ) : null;
}

const StyledFieldset = styled.fieldset`
  border: 0;
  margin: 0;
  padding: 0;
`;

function Fieldset({ additionalLegendClasses, legendText, children, theme }) {
  return (
    <ThemeProvider theme={theme}>
      <StyledFieldset>
        {renderLegend(legendText, additionalLegendClasses)}
        {children}
      </StyledFieldset>
    </ThemeProvider>
  );
}

Fieldset.propTypes = {
  /** Determine whether or not to add a legend and what text to display  */
  legendText: PropTypes.object,
  /** Additional classes to be applied to the legend element */
  additionalLegendClasses: PropTypes.string,
  children: PropTypes.node,
  /**
   * Theme object used by the ThemeProvider,
   * automatically passed by any parent component using a ThemeProvider
   */
  theme: PropTypes.object
};

Fieldset.defaultProps = {
  theme: defaultTheme
};

export default Fieldset;
