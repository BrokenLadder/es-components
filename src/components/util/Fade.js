import React from 'react';
import PropTypes from 'prop-types';
import Transition from 'react-overlays/lib/Transition';
import { injectGlobal } from 'styled-components';

/* eslint-disable no-unused-expressions */
injectGlobal`
  .default-transition-out {
    opacity: 0;
    transition: opacity 300ms linear;
  }

  .default-transition-in {
    opacity: 1;
  }
`;
/* eslint-enable no-unused-expressions */

const Fade = props => {
  const {
    in: transitionIn,
    timeout,
    transitionClassIn,
    transitionClassOut,
    ...otherProps
  } = props;

  return (
    <Transition
      className={transitionClassOut}
      enteredClassName={transitionClassIn}
      enteringClassName={transitionClassIn}
      in={transitionIn}
      unmountOnExit
      timeout={timeout}
      transitionAppear
      {...otherProps}
    />
  );
};

Fade.propTypes = {
  children: PropTypes.any.isRequired,
  in: PropTypes.bool,
  transitionClassIn: PropTypes.string,
  transitionClassOut: PropTypes.string,
  timeout: PropTypes.number
};

Fade.defaultProps = {
  transitionClassIn: 'default-transition-in',
  transitionClassOut: 'default-transition-out',
  timeout: 5000
};

export default Fade;
