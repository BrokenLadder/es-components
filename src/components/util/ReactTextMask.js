import React from 'react';
import ReactTextMask from 'react-text-mask';

// the component was using onInput which made testing more difficult.
// switched to onChange and put in a pull request so hopefully we can
// get rid of this soon.
ReactTextMask.prototype.render = function render() {
  const props = { ...this.props };

  delete props.mask;
  delete props.guide;
  delete props.pipe;
  delete props.placeholderChar;
  delete props.keepCharPositions;
  delete props.onChange;
  delete props.value;
  delete props.showMask;

  return (
    <input
      {...props}
      onChange={event => this.onChange(event)}
      defaultValue={this.props.value}
      ref={inputElement => (this.inputElement = inputElement)}
    />
  );
};

export default ReactTextMask;
