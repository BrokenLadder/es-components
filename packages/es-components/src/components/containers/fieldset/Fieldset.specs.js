/* eslint-env jest */

import React from 'react';
import { renderWithTheme } from '../../util/test-utils';

import Fieldset from './Fieldset';

it('renders legend when legendContent is provided', () => {
  const { queryByText } = renderWithTheme(
    <Fieldset legendContent="I am legend">
      <div>Fieldset child</div>
    </Fieldset>
  );

  expect(queryByText('I am legend')).not.toBeNull();
});
