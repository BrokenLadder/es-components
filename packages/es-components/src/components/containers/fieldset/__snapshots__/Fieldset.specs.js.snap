// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset component renders as expected 1`] = `
<fieldset
  className="es-fieldset sc-bwzfXH kzYDuJ"
>
  <div>
    Fieldset child
  </div>
</fieldset>
`;

exports[`Fieldset component renders as expected with legendText 1`] = `
<fieldset
  className="es-fieldset sc-bwzfXH kzYDuJ"
>
  <legend
    className="es-fieldset__legend sc-bdVaJa NnbOj"
  >
    I am legend
  </legend>
  <div>
    Fieldset child
  </div>
</fieldset>
`;
